File: tmp/bf.lua

w=[[
++++++++[>++++[>++>+++>+++>+<<<<-]>+>+>->>+[<]<-]>>.>---.+++++++..+++.>>.<-.<.+++.------.--------.>>+.>++.
]]

p,k,t=1,1,{}r={}
while p < #w do
    v=(t[k]or 0)z=string.byte(w,p)p=p+1
    if u then
            if 91 == z then u=u+1
        elseif 93 == z then u=0~=u and u-1
        end
    else
            if 43 == z then t[k]=v+1
        elseif 44 == z then t[k]=string.byte(io.read(1))
        elseif 45 == z then t[k]=v-1
        elseif 46 == z then io.write(string.char(v))
        elseif 60 == z then k=k-1
        elseif 62 == z then k=k+1
        elseif 91 == z then if 0~=v then r[#r+1]=p-1 else u=0 end
        elseif 93 == z then p,r[#r]=r[#r]
        end
    end
end

File: tmp/bf.js

(([w], t=[], k=0, r=((w, r, z, v, p=0)=>{
    while (v=t[k]||0,z=w[p++]?.charCodeAt()) -{
              43: _ => t[k]=v+1
            , 44: _ => (require(([]+!1)[0]+([]+!1)[3]).readSync(0,b=Buffer.alloc(1),0,1),t[k]=b.toString().charCodeAt())
            , 45: _ => t[k]=v-1
            , 46: _ => process.stdout.write(String.fromCharCode(v))
            , 60: _ => --k
            , 62: _ => ++k
            , 91: _ => p+=r(w.slice(p),r)
            , 93: _ => p=v?0:(r=p,NaN)
        }[z]?.()
    return r
})) => r(w, r))`
++++++++[>++++[>++>+++>+++>+<<<<-]>+>+>->>+[<]<-]>>.>---.+++++++..+++.>>.<-.<.+++.------.--------.>>+.>++.
`
